File: samples/customer-registry-quickstart/src/main/java/customer/view/CustomerByNameView.java
/* This code was initialised by Akka Serverless tooling.
 * As long as this file exists it will not be re-generated.
 * You are free to make changes to this file.
 */

package customer.view;

import com.akkaserverless.javasdk.view.ViewCreationContext;
import customer.domain.CustomerDomain;

/** A view. */
public class CustomerByNameView extends AbstractCustomerByNameView {

    public CustomerByNameView(ViewCreationContext context) {
    }

    @Override
    public CustomerDomain.CustomerState emptyState() {
        return null;
    }

    @Override
    public UpdateEffect<CustomerDomain.CustomerState> updateCustomer(CustomerDomain.CustomerState state, CustomerDomain.CustomerState customerState) {
        return updateEffects().updateState(customerState);
    }
}

File: samples/scala-eventsourced-customer-registry-subscriber/src/main/scala/customer/view/CustomerByNameView.scala
package customer.view

import customer.api.Created
import customer.api.Customer
import customer.api.NameChanged
import kalix.scalasdk.view.View.UpdateEffect
import kalix.scalasdk.view.ViewContext

// This class was initially generated based on the .proto definition by Kalix tooling.
//
// As long as this file exists it will not be overwritten: you can maintain it yourself,
// or delete it so it is regenerated as needed.

/** not currently working
class CustomerByNameView(context: ViewContext) extends AbstractCustomerByNameView {

  override def emptyState: Customer = Customer.defaultInstance

  override def processCustomerCreated(
    state: Customer, created: Created): UpdateEffect[Customer] =
    effects.updateState(state.copy(customerId = created.customerId, email = created.email, name = created.customerName))

  override def processCustomerNameChanged(
    state: Customer, nameChanged: NameChanged): UpdateEffect[Customer] =
    effects.updateState(state.copy(name = nameChanged.customerName))
}
  */
